code/src/components/PostMessage.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+
+import { HAPPY_THOUGHTS_API } from '../reusable/urls'
+
+const PostMessage = ({ messageNew, setMessageNew, messageList, setMessageList, chars, setChars }) => {
+
+  const onMessageNewChange = (event) => {
+    setMessageNew(event.target.value)
+    setChars(event.target.value.length)
+  }
+
+  const fetchMessageList = () => {
+    fetch(HAPPY_THOUGHTS_API)
+      .then(response => response.json())
+      .then(messages => setMessageList(messages))
+      .catch(err => console.error(err))
+  }
+
+  const onFormSubmit = (event) => {
+    event.preventDefault()
+
+    const postArgument = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ message: messageNew })
+    }
+
+    fetch(HAPPY_THOUGHTS_API, postArgument)
+      .then(response => response.json())
+      .then(() => {
+        setMessageNew('')
+        fetchMessageList()})
+  }
+
+  return (
+    <article className="message-container form-container">
```

Comment:
I would consider the usage of an article tag here. As explained in my comment on Header.js, article tags are generally intended for forum posts, newspaper articles, blog entries etc. If you need a wrapper here I would consider using a section element instead 🙂